include/zephyr/llext/llext.h

```diff
@@ -88,19 +101,22 @@ struct llext *llext_by_name(const char *name);
  * @param[in] loader An extension loader that provides input data and context
  * @param[in] name A string identifier for the extension
  * @param[out] ext A pointer to a statically allocated llext struct
+ * @param[in] do_local Perform local relocations in addition to external linking
  *
  * @retval 0 Success
  * @retval -ENOMEM Not enough memory
  * @retval -EINVAL Invalid ELF stream
  */
-int llext_load(struct llext_loader *loader, const char *name, struct llext **ext);
+int llext_load(struct llext_loader *loader, const char *name, struct llext **ext, bool do_local);
```

Comment:
I feel like we're likely to have more various options/flags/etc for configuring how loading and linking occurs, can we move the do_local bool the loader struct itself?

Reply:
The `do_local` argument seems to be only used to prevent `llext_link_plt` from calling `arch_elf_relocate_local`, which is useful for Xtensa but has a weak stub for other arches. Is this argument really needed?

Reply:
@teburd I'm trying to limit the scope of variables. This parameter is only needed within this single loading call. We now store the llext context permanently - from loading until unloading, but this parameter is only needed during loading. If we get more of such short-lived parameters, I'd maybe add a temporary loader-context structure, that isn't stored permanently and thus is just an object on stack in `llext_load()`